### PR TITLE
fix(insured-bridge): clean unit tests and simplify param injection

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

In writing the bridge pool and insured bridge clients, some code was made somewhat repetitive due to the nature of the contract interfaces. This PR introduces some helper functions that aim to clean up some unit tests (see code diff of lines removed).

**Summary**

Introduces helper methods the build function params to decrease LOC.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [X]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+    // Fetch event information
+    // TODO: consider optimizing this further. Right now it will make a series of sequential BlueBird calls for each pool.
+    for (const l1Token of Object.keys(this.bridgePools)) {
+      const [
+        depositRelayedEvents,
+        relaySpedUpEvents,
+        relayDisputedEvents,
+        relaySettledEvents,
+      ] = await BluebirdPromise.all([
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:122]: mainly done just for consistency with the other clients. we might want to add in rate-limiting down the road.

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        this.bridgePools[l1Token].getPastEvents("DepositRelayed", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelaySpedUp", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelayDisputed", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelaySettled", blockSearchConfig),
+      ]);
+
+      for (const depositRelayedEvent of depositRelayedEvents) {
+        const relayData = {
+          depositId: Number(depositRelayedEvent.returnValues.depositId),
+          sender: depositRelayedEvent.returnValues.sender,
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:132]: ye, that's a great point. I did it this way for the ordering of the keys in the object but there is not really a strong reason to do it this way. will spread the values rather.

[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:132]: one issue with doing it the way you specified is the logic below requires some of these values to be defined (such as `depositHash`). If this is not set then these methods have an issue with it. 

One work around is to set this `relayData` to the type `Relay` but then this assignment has a problem as not all the values are known until the promise returns. will leave it like this for now and work on making the types happy in the more concise technique.

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    }
+    this.firstBlockToSearch = blockSearchConfig.toBlock + 1;
+
+    this.logger.debug({
+      at: "InsuredBridgeL1Client",
+      message: "Insured bridge l1 client updated",
+    });
+  }
+
+  _throwIfNotInitialized() {
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:185]: done

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+  private firstBlockToSearch: number;
+
+  constructor(
+    private readonly logger: Logger,
+    readonly bridgeAdminAbi: Abi,
+    readonly bridgePoolAbi: Abi,
+    readonly l1Web3: Web3,
+    readonly bridgeAdminAddress: string,
+    readonly startingBlockNumber: number = 0,
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:49]: done

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  private relays: { [key: string]: { [key: string]: Relay } } = {}; // L1TokenAddress=>depositHash=>Relay.
+
+  private firstBlockToSearch: number;
+
+  constructor(
+    private readonly logger: Logger,
+    readonly bridgeAdminAbi: Abi,
+    readonly bridgePoolAbi: Abi,
+    readonly l1Web3: Web3,
+    readonly bridgeAdminAddress: string,
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:48]: hhmm. ye I don think they should be modified and they should have pubic accessors. I was really just following the pattern laid out in the other clients. is there something different you'd recommend?

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  async update(): Promise<void> {
+    // Define a config to bound the queries by.
+    const blockSearchConfig = {
+      fromBlock: this.firstBlockToSearch,
+      toBlock: this.endingBlockNumber || (await this.l2Web3.eth.getBlockNumber()),
+    };
+    // TODO: update this state retrieval to include looking for L2 liquidity in the deposit box that can be sent over
+    // the bridge. This should consider the minimumBridgingDelay and the lastBridgeTime for a respective L2Token.
+    const [depositRelayedEvents] = await BluebirdPromise.all([
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:62]: removed for now.

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      this.bridgeDepositBox.getPastEvents("FundsDeposited", blockSearchConfig),
+    ]);
+
+    for (const depositRelayedEvent of depositRelayedEvents) {
+      this.deposits[depositRelayedEvent.returnValues.depositId] = {
+        depositId: Number(depositRelayedEvent.returnValues.depositId),
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:68]: similar to the other client, I had type issues when doing this. I'm not sure how to make ts happy with this format. This is the error I get (and the modified definition of the structure):

image

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+    this.firstBlockToSearch = startingBlockNumber;
+  }
+
+  getAllDeposits() {
+    return Object.keys(this.deposits).map((depositId: string) => this.deposits[depositId]);
+  }
+
+  getDepositByID(depositId: string | number) {
+    return this.deposits[String(depositId)];
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts:51]: done

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+      // Else, we set the bridge pool to be a contract instance at the address of the bridge pool.
+      this.bridgePools[l1Token] = (new this.l1Web3.eth.Contract(
+        this.bridgePoolAbi,
+        whitelistedTokenEvent.returnValues.bridgePool
+      ) as unknown) as BridgePoolWeb3;
+      this.relays[l1Token] = {};
+    }
+
+    // Fetch event information
+    // TODO: consider optimizing this further. Right now it will make a series of sequential BlueBird calls for each pool.
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:115]: ye, that's pretty much the workaround I was going to use. I think it's best for us to not do this now to keep this easy to reason about. we can optimize later, if need be

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+    for (const l1Token of Object.keys(this.bridgePools)) {
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:116]: that is clean! good recomend

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrice32) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts]: Totally fine to leave it as is if it's annoying. I realized after I made the comment that you need to do an annoying dance to get this sort of assignment to work, which is probably not worth it.

File: packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (mrice32) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts]: Makes sense!

**Pull request merged**